Subject: [PATCH] m68k: dnfb doesn't check for Apollo
Cc: linux-fbdev-devel@lists.sourceforge.net

From: Geert Uytterhoeven <geert@linux-m68k.org>

The Apollo frame buffer device driver (dnfb) doesn't check whether it's
actually running on Apollo hardware, causing a crash if it isn't.

Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
---
 drivers/video/dnfb.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/video/dnfb.c
+++ b/drivers/video/dnfb.c
@@ -284,6 +284,9 @@ int __init dnfb_init(void)
 {
 	int ret;
 
+	if (!MACH_IS_APOLLO)
+		return -ENODEV;
+
 	if (fb_get_options("dnfb", NULL))
 		return -ENODEV;
 
